### ISSUE TITLE
Rename doesn't work

### ISSUE THREAD

**Issue by usernamehw** [microsoft/vscode#149028]
1. Register a rename provider with `pattern` specified

```ts
languages.registerRenameProvider({
	scheme: 'file',
	pattern: '**/*.md',
}, {
	provideRenameEdits(document, position) {
		const edit = new WorkspaceEdit();
		edit.insert(document.uri, position, '☑☑☑');
		return edit;
	},
});
```

2. Try to rename anything except header in a markdown file

> Rename not supported at location

---

There is a rename provider for headers in markdown, but any `pattern` matching `**/something.md` or `**/*.md` should produce a better match to override that (or work alongside it).

**Comment by jrieken** [comment]
> matching **/something.md or **/*.md should produce a better match to override that (or work alongside it).

No, they score equally well to just `markdown`. Please disable the other extension and try again

**Comment by usernamehw** [comment]
Disable the built-in markdown extension?

**Comment by jrieken** [comment]
yes
